WebRTC remote video codec factories should check video format strings in a case insensitive way

youennf opened the pull request:
#### 758b9fa876abfee20303553bcdb17de2b92746e1
<pre>
WebRTC remote video codec factories should check video format strings in a case insensitive way
<a href="https://bugs.webkit.org/show_bug.cgi?id=261099">https://bugs.webkit.org/show_bug.cgi?id=261099</a>
rdar://114839348

Reviewed by Jean-Yves Avenard.

Make WebRTC codec checks case insensitive, which follows what other browsers are doing.
Add a WebRTC test case to cover this case.

Keep doing case sensitive checks for WebCodecs and beef up WPT webcodecs tests accordingly.

* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any.worker-expected.txt:
* LayoutTests/webrtc/video-lowercase-media-subtype-expected.txt: Added.
* LayoutTests/webrtc/video-lowercase-media-subtype.html: Added.
* LayoutTests/platform/glib/TestExpectations:
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
(WebKit::createVideoDecoder):
(WebKit::createVideoEncoder):

Canonical link: <a href="https://commits.webkit.org/267637@main">https://commits.webkit.org/267637@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/638a6ff7314e5d401f47834587a4fc87a0790eb9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17173 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17499 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17993 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18960 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16071 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20773 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17641 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18267 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17376 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17720 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14909 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19777 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14960 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15600 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22290 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15959 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15767 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20108 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16360 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13877 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15514 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/15440 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4120 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19881 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16191 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
```diff
@@ -65,16 +65,18 @@ using namespace WebCore;
 {
     auto& codecs = WebProcess::singleton().libWebRTCCodecs();
-    if (format.name == "H264")
+    auto codecString = String::fromUTF8(format.name.data(), format.name.length());
```
no VP8 ?

youennf replied:
Not needed since libvpx is used for VP8 and is done in process.
createVideoDecoder is for HW based decoders

eric-carlson commented on Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
```diff
@@ -124,10 +126,10 @@ static int32_t registerDecodeCompleteCallback(webrtc::WebKitVideoDecoder::Value
 static webrtc::WebKitVideoEncoder createVideoEncoder(const webrtc::SdpVideoFormat& format)
 {
-    if (format.name == "H264")
+    if (format.name == "H264" || format.name == "h264")
```
Is there a reason to use `equalIgnoringASCIICase` in `createVideoDecoder` and test both variants explicitly here?

youennf replied:
It is a minor optimisation, we do not need to copy the string in WTFString to use equalIgnoringASCIICase. We cannot do this trick in the other cases because of 'VP9' and 'AV1' 4 variants.

youennf committed:
WebRTC remote video codec factories should check video format strings in a case insensitive way
https://bugs.webkit.org/show_bug.cgi?id=261099
rdar://114839348

Reviewed by Jean-Yves Avenard.

Make WebRTC codec checks case insensitive, which follows what other browsers are doing.
Add a WebRTC test case to cover this case.

Keep doing case sensitive checks for WebCodecs and beef up WPT webcodecs tests accordingly.

* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-decoder.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any.js:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/video-encoder-config.https.any.worker-expected.txt:
* LayoutTests/webrtc/video-lowercase-media-subtype-expected.txt: Added.
* LayoutTests/webrtc/video-lowercase-media-subtype.html: Added.
* LayoutTests/platform/glib/TestExpectations:
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
(WebKit::createVideoDecoder):
(WebKit::createVideoEncoder):

Canonical link: https://commits.webkit.org/267637@main
